### PR TITLE
IR-471: Removing featuregate for chunkSizeMiB config

### PR DESCRIPTION
**Pull request by deepsm007**
/cc @flavianmissi https://github.com/openshift/cluster-image-registry-operator/pull/1073 

[Sippy test runs](https://sippy.dptools.openshift.org/sippy-ng/tests/4.17?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522never-stable%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522aggregated%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Chunksizemib%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&sort=asc&sortField=current_working_percentage)

Tested with cluster-bot using [this](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/release-openshift-origin-installer-launch-aws-modern/1819091147578937344)

```
cluster-bot]$ oc get clusterversion version
NAME      VERSION                                                AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.17.0-0.test-2024-08-01-193338-ci-ln-xxm7wdb-latest   True        False         5m24s   Cluster version is 4.17.0-0.test-2024-08-01-193338-ci-ln-xxm7wdb-latest

cluster-bot]$ oc patch config.image/cluster -p '{"spec":{"storage":{"s3":{"chunkSizeMiB":5}}}}' --type=merge
config.imageregistry.operator.openshift.io/cluster patched

cluster-bot]$ oc -n openshift-image-registry get  config.image/cluster -oyaml | yq .spec.storage.s3
bucket: ci-ln-xxm7wdb-76ef8-77pmb-image-registry-us-east-2-trhxwyqtpsk
chunkSizeMiB: 5
encrypt: true
region: us-east-2
trustedCA:
  name: ""
virtualHostedStyle: false
```

### CODE REVIEW

**Comment by flavianmissi** [comment]
LGTM to me from the registry side
/lgtm

**Comment by flavianmissi** [comment]
Something is off... from `e2e-aws-ovn` [job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_api/1948/pull-ci-openshift-api-master-e2e-aws-ovn/1816583115309584384):
```
Observed a panic: &errors.errorString{s:"feature \"ChunkSizeMiB\" is not registered in FeatureGates
```

File: features/features.go [hunk 1]
```diff
@@ -527,7 +527,6 @@ var (
 				reportProblemsToJiraComponent("Image Registry").
 				contactPerson("flavianmissi").
 				productScope(ocpSpecific).
-				enableIn(configv1.DevPreviewNoUpgrade, configv1.TechPreviewNoUpgrade).
```
Comment (JoelSpeed) [inline features/features.go:530]: This should not be removed, just add `configv1.Default` as an additional item in the input

File: imageregistry/v1/types.go [hunk 2]
```diff
@@ -213,7 +213,6 @@ type ImageRegistryConfigStorageS3 struct {
 	// The minimum value is 5 and the maximum value is 5120 (5 GiB).
 	// +kubebuilder:validation:Minimum=5
 	// +kubebuilder:validation:Maximum=5120
-	// +openshift:enable:FeatureGate=ChunkSizeMiB
```
Comment (JoelSpeed) [inline imageregistry/v1/types.go:216]: Don't do this just yet please

**Comment by deepsm007** [comment]
[Sippy test runs](https://sippy.dptools.openshift.org/sippy-ng/tests/4.17?filters=%257B%2522items%2522%253A%255B%257B%2522columnField%2522%253A%2522current_runs%2522%252C%2522operatorValue%2522%253A%2522%253E%253D%2522%252C%2522value%2522%253A%25227%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522never-stable%2522%257D%252C%257B%2522columnField%2522%253A%2522variants%2522%252C%2522not%2522%253Atrue%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522aggregated%2522%257D%252C%257B%2522id%2522%253A99%252C%2522columnField%2522%253A%2522name%2522%252C%2522operatorValue%2522%253A%2522contains%2522%252C%2522value%2522%253A%2522Chunksizemib%2522%257D%255D%252C%2522linkOperator%2522%253A%2522and%2522%257D&sort=asc&sortField=current_working_percentage)

**Comment by JoelSpeed** [comment]
/override ci/prow/verify-crd-schema 

The only failures here are because of maps and bools, and cannot be fixed.

/override ci/prow/verify

Based on the sippy test runs above, I'm also happy that this is being tested and the verify checks (complaining that this feature doesn't work on Azure) are not something that makes sense in this context

One quick question, are there any other opinions we want to gather before we promote? Perhaps adding QE ack would be good?

**Comment by deepsm007** [comment]
> One quick question, are there any other opinions we want to gather before we promote? Perhaps adding QE ack would be good?

QE will be testing this before we can confirm to merge the PR. cc @xiuwang is working on this and i'll wait for her feedback

**Comment by JoelSpeed** [comment]
```
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should reject ChunkSizeMiB value greater than 5 GiB [Suite:openshift/conformance/parallel/minimal]" only passed 76%, need at least 95% for "ChunkSizeMiB" on {azure amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should set ChunkSizeMiB value [Suite:openshift/conformance/parallel/minimal]" only passed 84%, need at least 95% for "ChunkSizeMiB" on {azure amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should set maximum valid ChunkSizeMiB value [Suite:openshift/conformance/parallel/minimal]" only passed 76%, need at least 95% for "ChunkSizeMiB" on {azure amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should set minimum valid ChunkSizeMiB value [Suite:openshift/conformance/parallel/minimal]" only passed 73%, need at least 95% for "ChunkSizeMiB" on {azure amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should not accept invalid ChunkSizeMiB value [Suite:openshift/conformance/parallel/minimal]" only passed 80%, need at least 95% for "ChunkSizeMiB" on {gcp amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should reject ChunkSizeMiB value greater than 5 GiB [Suite:openshift/conformance/parallel/minimal]" only passed 84%, need at least 95% for "ChunkSizeMiB" on {gcp amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should set ChunkSizeMiB value [Suite:openshift/conformance/parallel/minimal]" only passed 68%, need at least 95% for "ChunkSizeMiB" on {gcp amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should set maximum valid ChunkSizeMiB value [Suite:openshift/conformance/parallel/minimal]" only passed 88%, need at least 95% for "ChunkSizeMiB" on {gcp amd64 ha}
error: "[sig-imageregistry][OCPFeatureGate:ChunkSizeMiB][Conformance][apigroup:imageregistry.operator.openshift.io] Image Registry Config ChunkSizeMiB should set minimum valid ChunkSizeMiB value [Suite:openshift/conformance/parallel/minimal]" only passed 80%, need at least 95% for "ChunkSizeMiB" on {gcp amd64 ha}
```

Looks like we are picking up the test results correctly now. Percentages are lower than we require to pass automatically but you have QE sign off so I'm happy to merge

/lgtm
/override ci/prow/verify-crd-schema 
/override ci/prow/verify

**Comment by JoelSpeed** [comment]
/override ci/prow/verify
/override ci/prow/verify-crd-schema 

Tests are being skipped temporarily to avoid an ordering issue with merging this, but not updating payload CRDs directly. Tests need to be re-enabled once the CRDs are updated in payload.

See https://github.com/openshift/origin/pull/28975 for context

**Pull request merged**